src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -102,16 +131,45 @@ export class GridOutputComponent extends AngularDisposable implements IMimeCompo
 		if (!this._table) {
 			let source = <IDataResource><any>this._bundleOptions.data[this.mimeType];
 			let state = new GridTableState(0, 0);
-			this._table = this.instantiationService.createInstance(DataResourceTable, source, this.cellModel, this.cellOutput, state);
+			this._table = this.instantiationService.createInstance(DataResourceTable, this._batchId, this._id, this._queryRunner, source, this.cellModel, this.cellOutput, state);
 			let outputElement = <HTMLElement>this.output.nativeElement;
 			outputElement.appendChild(this._table.element);
 			this._register(attachTableStyler(this._table, this.themeService));
 			this._table.onDidInsert();
 			this.layout();
+			if (this._queryRunner) {
+				this._register(this._queryRunner.onResultSetUpdate(resultSet => { this.updateResultSet(resultSet); }));
+				this._register(this._queryRunner.onBatchEnd(batch => { this.convertData(batch); }));
+			}
 			this._initialized = true;
 		}
 	}
 
+	updateResultSet(resultSet: ResultSetSummary | ResultSetSummary[]): void {
+		let resultsToUpdate: ResultSetSummary[];
+		if (!Array.isArray(resultSet)) {
+			resultsToUpdate = [resultSet];
+		} else {
+			resultsToUpdate = resultSet.splice(0);
```

Comment:
what does `splice(0)` do in this context? Is there a case I'm not thinking of?

Reply:
splice(0) removes the contents in resultSet and copies it into resultToUpdate. I copied this from GridPanel so @anthonydresser probably has a better answer?

Reply:
It clones the array, otherwise you can be modifying the original array held by the query runner (`.splice()` does the same thing). I think the return type of this from the QueryRunner is `ReadonlyArray` which means if you want to modify it, you have to clone it.